### ISSUE TITLE
Merge forums, OSQA, MLs to discourse?

### ISSUE THREAD

**Issue by Matthias84** [openstreetmap/operations#377]
At #149 we discuss a alternative to OSQA and some already mentioned [discourse](https://www.discourse.org/) based forums, which are used by many other FLOSS communities as well.

This step might also allow us to unite different OSM community channels, which some see as ongoing problem for newbies (_which one to choose?_) and long time contributors (_monitor them all, splitted discussions, ..._)

Discourse offers possibilities to import discussions from
*  [for OSQA](https://github.com/HerbJiang/discourse-import-osqa), 
*  [for fluxBB](https://meta.discourse.org/t/importing-from-fluxbb/51322)
* [for mailinglists](https://meta.discourse.org/t/importing-mailing-lists-mbox-listserv-google-groups-emails/79773)

Would this be a useful second step, if we get a discourse setup running?

**Comment by rory** [comment]
Is it possible to use discourse as a mailing list? i.e. to interact with it via email? If not, then you'll never get rid of the mailing lists, and instead you'll just add an additional communication channel.

Have you discussed this with existing heavy OSM Forum users? Do they have any feedback on what Discourse is like compared to the existing OSM Forum software.

As someone who cannot log into the OSM Forum because the software doesn't support my OSM username, changing the software has advantages to me.

**Comment by tomhughes** [comment]
It has "mailing list mode" yes.

I mean obviously it's not the same as a real mailing list but it's probably the closest anybody has ever come to creating an integrated forum/mailing list system.

**Comment by Nakaner** [comment]
The other option is to use Mailman 3 which can [use authentication via OpenStreetMap (OAuth)](https://lists.openrailwaymap.org/mailman3/accounts/login/?next=/mailman3/accounts/signup/) and has a simple web frontend to send emails. It is in use by OpenRailwayMap. [Try it out.](https://lists.openrailwaymap.org/mailman3/postorius/lists/mailman3-sandbox.openrailwaymap.org/)

OpenRailwayMap uses Debian packages (but self built packages with slightly more recent versions and some patches). The packaging can be found at

* https://github.com/fossgis/django_mailman3-debian
* https://github.com/fossgis/hyperkitty-debian
* https://github.com/fossgis/postorius-debian
* https://github.com/fossgis/mailman3-debian

The ansible playbook is [available](https://github.com/OpenRailwayMap/server-admin/blob/master/ansible/roles/mail/tasks/main.yml#L292) as well.

**Comment by tomhughes** [comment]
Literally nobody on earth is actually using mailman 3 other than Fedora.

Well OK apparently you've found the one other place that does ;-)

In addition to apparently being a massive PITA to install (though I expect discourse is as well...) it also has a horrible UI to boot.

Even Fedora has now spun up a [discourse](https://ask.fedoraproject.org/) instance though the internal lists are still using mailman 3 for now.

**Comment by Matthias84** [comment]
sorry folks, but please **no off-topic** and keep discussions on pros / cons of discourse at ticket #149 but not here. This is dedicated about merging our internal communication channels.

Let's find out, if unified forums is a chance for our community?

**Comment by pnorman** [comment]
> Have you discussed this with existing heavy OSM Forum users? Do they have any feedback on what Discourse is like compared to the existing OSM Forum software.

I think we need to reach out to those people to get them involved in the discussion.

**Comment by Matthias84** [comment]
I started topics [at forum](https://forum.openstreetmap.org/viewtopic.php?pid=780678) and [at OSQA](https://help.openstreetmap.org/questions/73580/switch-to-discourse). My idea is to wait for the WeeklyOSM, but of course anybody can start discussions at the talk ML or other useful places?

**Comment by pnorman** [comment]
cc @tordanik

**Comment by gravitystorm** [comment]
> This is dedicated about merging our internal communication channels.

FWIW, I'm opposed to merging the forums and the help Q&A site. We definitely have problems with the underlying OSQA software, but the two venues serve completely different use cases.

More generally, I don't think that merging disparate communication channels has a significant benefit. With an ever-expanding community, it's foolish to attempt (or to prioritise) the ability to "monitor all the communications", and different people and different groups of people prefer different styles of communication in different situations (real-time chat, questions and answers, rambling discussions, short-form message broadcasting, long form blogging, etc etc). There's no pressing requirement to combine these different formats together. 

So I think before attempting to merge anything, it needs more careful analysis about the style and purpose of communication in each format, and the current and future volumes of communication in the combined format, and whether this then makes sense. I would expect, for example, even a simple analysis to show that the mailing lists and help Q&A to have little overlap in either purpose or style of communication.

**Comment by mrtngrsbch** [comment]
FluxBB and Discourse are not 'completely different' but a technological evolution to the demand of users and administrators.
On the other hand, our Telegram channels are overloaded with chaotic information and very difficult to read and follow.

The forum of the Argentinean community is in 4th place in number of posts and only contains a tiny part of the information that circulates through other channels.

Canada has 38 topics, Spain 91...
It does not take much study to realize that the forum is a cemetery and the different communities are looking for other ways. Moreover, there is already a Discord World!

Times have changed and we must understand the new challenges and offer solutions before the world imposes them.

In Argentina we have created a specific working group for this topic and we are studying all the possibilities to improve the general communication, mainly for the new users who have many difficulties to follow the diversity of channels that we currently have [https://wiki.openstreetmap.org/wiki/ES:Argentina#Contacto].

Currently, our main channel is [https://telegram.me/osm_ar] but we also have themes [Covid, Nature, GLAM] that were created to ease the charge.

Indeed, Telegram is not the appropriate tool for this type of problem, so we are seriously considering migrating the Forum to the Discourse. In this way we hope to be able to channel general questions or queries to the forum and to retain Telegram for more fluid conversations.

We are also considering Telefuel where we already have a Beta installation [https://www.telefuel.com/w/001Jdh8Mw4sgXsTpezCK/] so we can have 'channels' and not migrate quickly to Discord.

In short, to this day we think:
- Preserve the Forum as a historical channel but we would migrate the posts.
- We would make a more complete homepage with a CMS [https://openstreetmap.org.ar/]
- We would keep the Telegram, for the time being

best

**Comment by mrtngrsbch** [comment]
> There is user [&ergio](https://forum.openstreetmap.org/viewtopic.php?pid=782859#p782859) which is willing to help on a discourse server!

I just asked the forum admins to offer me an export to import into Discourse...
In the Argentine community we are willing to be the laboratory of this idea
[[https://wiki.openstreetmap.org/wiki/Talk:Forum](https://wiki.openstreetmap.org/wiki/Talk:Forum)]

**Comment by Zverik** [comment]
>  the forum is a cemetery and the different communities are looking for other ways.

Uhm... The forum is the main communication channel for communities in Austria, Belarus, Netherlands, Germany, Russia, Poland, Ukraine and many other countries. If that's not the case for Argentina, that doesn't mean it isn't for other countries. If you contact active members of these communities, I doubt the decision to move to Discourse would pass that easy.

**Comment by mrtngrsbch** [comment]
I have no doubt that many users will want to continue with the forum and each of them must have their reasons, mainly out of habit.

But if we want to move forward we will have to take some action according to the 21st century

I was just looking at yesterday's stats from our main Telegram: 100 posts!
In the forum: 1

Also... it is not my idea to close the forum! We can leave it to the dinosaurs to play in their park ;-)

Seriously... I just found this thread in the forum [[https://forum.openstreetmap.org/viewtopic.php?pid=783989](https://forum.openstreetmap.org/viewtopic.php?pid=783989)] !
It seems that we are not the only ones looking to leave the cemetery.

**Comment by 532910** [comment]
> There is user [&ergio](https://forum.openstreetmap.org/viewtopic.php?pid=782859#p782859) which is willing to help on a discourse server!

It's me.

**Comment by 532910** [comment]
Илья, have you ever seen or used some of Discourse instances (don't confuse with Discord)?
It's used on many opensource projects, and it's definitely the best forum engine. So I believe there should be no objection from other members.

**Comment by rory** [comment]
Let's not berate people who use the forum. Language like “dinosaurs” doesn't help. (Plus dinosaurs are cool).

**Comment by mrtngrsbch** [comment]
> Let's not berate people who use the forum. Language like “dinosaurs” doesn't help. (Plus dinosaurs are cool).

Hi @rory,
I love dinosaurs ;-)
I'm a 53-year-old dinosaur, who learned computing with a Sinclair Z80!

The problem we have in AR and many other communities is that the forum is not used by anyone!
I don't have any problem with the forum, but it doesn't respond to users' needs, except for dinosaurs ;-)

In other words, we have to find a solution for the new users somehow.

**Comment by Matthias84** [comment]
PLz. stay ontopic. This is about the idea to shift to discourse and maybe to merge different existing channels into it.

**Comment by mmd-osm** [comment]
In case you missed it: https://wiki.osmfoundation.org/wiki/Operations/Minutes/2020-07-15#FluxBB_is_old_and_slow

Action item: @iandees to try converting fluxBB DB to go into Discourse.

**Comment by natrius** [comment]
In the german forum there is a thread as well https://forum.openstreetmap.org/viewtopic.php?id=68944

**Comment by mrtngrsbch** [comment]
> In the german forum there is a thread as well https://forum.openstreetmap.org/viewtopic.php?id=68944

It seems to me that it only contributes to the general confusion.
The first step I think should be to offer an alternative to the forum

**Comment by mmd-osm** [comment]
OAuth 1.0a vs. OAuth 2.0 seems to be one of the issues according to one of the recent forum posts.

Not sure if there are any plans to move OSM website to OAuth 2.0 or if writing a custom OAuth 1.0a plugin for DIscourse is doable? There's also a somewhat stalled issue on OAuth 2.0 for the OSM website out there:  https://github.com/openstreetmap/openstreetmap-website/issues/1408

**Comment by tomhughes** [comment]
I'm currently investigating options for OAuth 2 support.

**Comment by tomhughes** [comment]
As far as Discourse is concerned - can you confirm if OAuth 2.0 is sufficient? or does it need OpenID Connect?

**Comment by iandees** [comment]
Discourse has built in support for Oauth2: https://meta.discourse.org/t/oauth2-basic-support/33879

I can also use something like what you did for the existing forum, @tomhughes, but Oauth2 would obviously be better.

**Comment by tomhughes** [comment]
Right so no need for OpenID Connect but we would need JSON support on the user details API endpoint.

**Comment by mmd-osm** [comment]
> but we would need JSON support on the user details API endpoint.

My proposal would be something along these lines: https://github.com/mmd-osm/openstreetmap-website/commit/0fb725ea611e7788a05bcb39bbf65ccdfe7ae505

**Comment by HarelM** [comment]
We, In israel, have stopped using the forum completely and are using telegram now. We are a small community. The forum is not working well for us and of I need to guess it's due to the fact that it doesn't work well on mobile.
I don't think telegram is ideal, mainly due to spam, and I'd love to try out something else.

**Comment by mrtngrsbch** [comment]
We in Argentina have practically abandoned the forum and are using Telegram with +10 groups and it is NOT the best solution. New users are lost, it is very difficult to follow conversations, it induces experienced users who do not have time to 'mute'. Currently I have 650 unreads on the main channel, etc.

Telegram is only good for talking in groups, but it is lousy for preserving historical threads and for seeking problems and solutions that are usually in the forums.
That's why we trust Discourse to do the job well.

best

**Comment by natrius** [comment]
@HarelM and @mrtngrsbch 

Just to say, the new default theme in the forum is air3 and thats mobile-friendly. Existing users have to change it by themself. Suggest, you give it a try. (Just in Case - Settings - Display - Styles - Air3). I thought i had discussed that with israelian users already, but it was iran, but it should not matter as they use telegram as well https://forum.openstreetmap.org/viewtopic.php?id=67507

I would suggest that you guys give Element a try as Chat. It is based on the Matrix protocol and one advantage is: You can register now and if the osm-server will be set up(or you want to set up your own server), you can still use your old account due to federation. The next advantage are bridges - Telegram users can still use Telegram in the room (and there is a room-search). But, thats actually nothing for here. Same as Telegram, Element is for Chatting, not for storing information longterm. There is an issue for this as well https://github.com/openstreetmap/operations/issues/380 and if you wanna try element https://element.io/ its there. If you have questions about that, hit me up on osm "Negreheb"

**Comment by kocio-pl** [comment]
Just a thought from my current experiences: Matrix network (for which Riot/Element is a client) is meant for strongly distributed communication and it deploys encryption pretty much everywhere, especially lately, so when you loose you personal backup of keys, you have no way to view the history. This makes it less suited for a public chat or a centralized system for the organization, however usability and features are quite nice.

**Comment by natrius** [comment]
Last thing i post here about matrix/element because this issue is for Forum/Help/Discourse: Public rooms are not encrypted, private per default yes. I don't think public rooms are going to be encrypted by default for the reason you said right here. The encryption thing is a element-choice btw., not matrix itself.

**Comment by Matthias84** [comment]
Thanks for sharing all of your impressions. IMHO we see the repeating pattern of different requirements (live vs. asyncronous, rooms vs. threading, ...). As you say, we don't try to replace chats, but of course you can start a different topic on uniting messaging (IRC, twitter, telegram, ...) for the OSM community as well :wink: 

It's good to hear, that there are prototypes to test a migration to discourse :+1:

**Comment by mrtngrsbch** [comment]
The (asynchronous) forums are very (very) important and my particular interest is in new users.
When you start with a new topic, you are very restless and have many questions that you ask a search engine and, thanks to it, you get the answers you were looking for.

Every week I have to learn new topics and you get that information from your findings [For example, yesterday I learned Scan Tailor, Tesseract and Translate Shell].

Leaving all interactions on Telegram is a deep wound to the transmission of knowledge.  New users are very important and we older users have the responsibility to transmit our knowledge in the best possible way and the world of OSM is too complex to deprive ourselves of such an important channel.

Now... I don't think the discussion is Discourse anymore, but how do we best implement it? Unfortunately in Argentina we do not have the human resources to carry it out, but I have no doubt that it will be welcome.

And I take this opportunity to reiterate my offer to use the Argentina (third in global volume) forum as a leading case.

**Comment by natrius** [comment]
What is needed right now, are there specific problems or obstacles? Or is it possible to set up somewhere on an alternative osm-subdomain a test-instance to check everything if someone is willing to try it?

**Comment by mmd-osm** [comment]
Last public status is here: https://wiki.osmfoundation.org/wiki/Operations/Minutes/2020-08-26#Merge_forums.2C_OSQA.2C_MLs_to_discourse.3F

**Comment by brocke** [comment]
> What is needed right now, are there specific problems or obstacles? Or is it possible to set up somewhere on an alternative osm-subdomain a test-instance to check everything if someone is willing to try it?

Yes exactly! To the admins and programmers: What is the current status, are there any major problems? When is it likely to be possible to test?
Thank you for working on this implementation, that will make the entire OSM project much more contemporary.

**Comment by tomhughes** [comment]
Current status is that the idea is being discussed on this ticket. No decision has been taken to go ahead, or what to go ahead with and no technical implementation work is currently being undertaken.

Depending on what exactly we decide this is going to be a major project and definitely not something that will happen overnight.

The main technical issue that needs to be resolved is integration with OSM accounts for authentication and I am working on that but it's a fairly large task so will take some time.

Then OWG would have to decide if they wanted to go ahead with discourse and if so whether to self host - integrated with that would probably have to be a large scale consultation exercise with current forum and mailing list users.

Basically this is something that, if it happens, is a project that will happen over months not days.

**Comment by sanecito** [comment]
I'd like to thrown in the suggestion of using Flarum over Discourse as it's also FOSS, but uses less system resources

**Comment by natrius** [comment]
It seems like Flarum cannot handle mailing-lists and this is, in my opinion, a major feature to reduce the communicationchannels from 3 to 1,5 :D But i think its always better to look at more options anyway.

(Thanks @sanecito for telling me, had both tabs opened :p )

**Comment by liotier** [comment]
I'm glad to stumble on this ticket - I was thinking about Discourse too, to solve some of our community issues. Do we have a shared document to build a common picture of the problems we face if we want to move this forward ?

**Comment by tomhughes** [comment]
No we're nowhere near actually being ready to try standing this up.

**Comment by RobJN** [comment]
>No we're nowhere near actually being ready to try standing this up.
What's needed @tomhughes? How can people help even if just getting us from "nowhere near" to "somewhere near"?

**Comment by tomhughes** [comment]
They can stop pestering me.

**Comment by Firefishy** [comment]
Minor announcement relevant to this ticket: https://twitter.com/OSM_Tech/status/1364718425407442949
We don't have a fixed timescale, but it is on my agenda.

**Comment by iandees** [comment]
I'm still ready to help in any way I can.

**Comment by matkoniecz** [comment]
What it would mean for mailing lists? Complete shutdown, people are invited to participate in a new forum? (seems to be aplan according to https://wiki.osmfoundation.org/wiki/Operations/Minutes/2020-08-26#Merge_forums.2C_OSQA.2C_MLs_to_discourse.3F )

> It has "mailing list mode" yes.

Where posts made via email are still possible and would be merged with posts made via web interface would be merged? (with some de-synchronization as edits to posts would be invisible to people who remain on mailing list interface).

**Comment by iandees** [comment]
> What it would mean for mailing lists?

Yes, my preference would be for archive/read-only on existing mailing lists (purposefully *not* migrating existing content).

Ideally, users could interact with Discourse only via email if they prefer that. It would become the new listserv for folks that want to keep things in their inbox. This would also let people search for posts more easily. I like it because moderation would be easier/possible.

**Comment by mrtngrsbch** [comment]
If possible, it would be ideal not to create new ones, but to respond to the posts created so far.

**Comment by rory** [comment]
> What it would mean for mailing lists? Complete shutdown, people are 
> invited to participate in a new forum? (seems to be aplan according to 

I would hope it would (from a mailing list point of view) just appear like the mailing list software has changed. You could just view it as “the mailing list software changed from Mailman 2 to Discourse”

On Thu, 11 Mar 2021 20:30 +01:00, Ian Dees ***@***.***> wrote:
> Ideally, users could interact with Discourse only via email if they 
> prefer that.

I would view that as a hard requirement, not a “in an ideal world”/“it would be nice to have”. IMO there *must* be a way for someone to continue to interact 100% over email (a once off requirement to create an account on a website is OK). We already have an email less web forum.

**Comment by iandees** [comment]
> IMO there *must* be a way for someone to continue to interact 100% over email

I'm curious: Why is email-only interaction important?

**Comment by matkoniecz** [comment]
> I'm curious: Why is email-only interaction important?

For me? I really dislike dealing with yet another interface, especially with JS heavy monster like Discourse (in comparison, Github issue threads are my favorite).

Discourse manages to lag on laptop with 16GB RAM. Load of page with text as sole relevant content takes seconds. On one of Discourse forums I lost today nontrivial post because I triggered some obscure functionality that deleted all what I have written. Badges are done in incredibly annoying way, the same for many other parts of its interface.

\<grumpy dinosaur mode off\>

Also, any transition to new interfaces causes some people to disappear forever, it would be nice to avoid it.

**Comment by rory** [comment]
On Thu, 11 Mar 2021 22:08 +01:00, Ian Dees ***@***.***> wrote:
> I'm curious: Why is email-only interaction important?

Because we currently have an active community of people who are using email only interface (i.e. the mailing lists). I'm sure some people would like if you could just turn off the mailing lists and tell that community “Use the web interface or get lost”, but let's be realistic. 

If it's not possible, it might still be useful for there to be an OSM Discourse site. It might be a nice interface. But in that case the mailing lists will still exist, and still be used. So it would not change a lot of things. There would still be all the technical downsides of the current mailing list software (some inherent to the lack of a proper web frontend)

**Comment by Matthias84** [comment]
As I pointed out, there is the discource feature to offer a email lists like interface (I guess similar to mailman3) and also the possibility to import existing mailinglist discussions.
Beside the technical limitations & problems of **merging forums AND mailinglists**. I don't read any points yet, which are con merging the lists with a forum? Has any mailinglist user tested discourse powered sites so far? What is your impression on usability, threading, .... ?

_(Don't get me wrong, I agree that there is the need to give people the prefered way to communicate with each other. For some this is plain email system, for others it's a webpage powered UI, for another one it's native mobile app or RSS integration. Let's see what might be covered by discourse. For alternative approaches, feel free to start a separate issue, so we can compare both)_

**Comment by rory** [comment]
On Fri, 12 Mar 2021 19:22 +01:00, Matthias84 ***@***.***> wrote:
> Has any mailinglist user tested discourse powered 
> sites so far? What is your impression on usability, threading, .... ?

I often use OSM mailing lists, and I am currently trialling a discourse based forum in mailing list mode (https://users.rust-lang.org/).

 * My email client (fastmail.com) doesn't do really do threading. But Discourse does group them all together.
 * It puts extra stuff at the bottom of the email (for links to the thread & post)
 * You don't get the senders email address, so you have to reply to the thread.
 * If you reply “+1” to a message, discourse records that as a “Like” on the post, which is cool
 * Admins need to turn on “allow threads to be created by email”, AFAIK it's off by default
 * Obv, you need to turn on things in the discourse settings for the first time.
 * The way it does 'topics'/'subfora' isn't as separate as different mailing lists
 * The web interface for disources is *miles* better than mailing lists
 * Discourse supports +1's for posts, which is a good way to see how many people agree with a mailing list post
 * Mailing lists just don't support searching (You need to hack it with a regular search engine)
 * Mailing lists have a big lacking feature used for moderation: You can't remove messages (like how in OSM we can redact data).

**Comment by rory** [comment]
On Thu, 11 Mar 2021 22:08 +01:00, Ian Dees ***@***.***> wrote:
> I'm curious: Why is email-only interaction important?

I'm curious, for those who think email-only interaction is optional: OSM has a forum, so what's wrong with that? Why use discourse?

**Comment by Matthias84** [comment]
The starting point is, that help.osm.org is outdated and doesn't get any more updates.
The forum is ok, but pretty limited compared to modern solutions and for non-tech users
* Image upload
* WYSIWYG Formating
* mobile apps
* ...

Bit wihtin this issues, pls lets focus on the topic of merging non realtime communication channels.

**Comment by pnorman** [comment]
We will continue to need an email-only way of interacting. If Discourse didn't support email, I doubt we would be considering it. We've had reports from multiple people about Discourse being used as a replacement for mailman2, which is what we use.

Even though we need to continue to support email-only, it's not going to be identical, and some things are going to be different than mailman 2. Some will be better, some will be worse.

For OSQA and FluxBB, the software powering the help and forum sites respectively, there are issues with both. OSQA is unmaintained and the [forks of it have died off](https://github.com/openstreetmap/operations/issues/149). We're maintaining a fork of FluxBB, so there are some issues there but no where near as bad as OSQA.

**Comment by SomeoneElseOSM** [comment]
> Why is email-only interaction important?

For me, it's because quite a lot of the time that I might be reading OSM-related stuff I may only have an intermittent network connection or none at all.  Not an issue while sat at home due to COVID 19 restrictions, obviously, but likely to become one again soon I hope.  I typically read lots of OSM stuff "offline" and then mark it if I want to do something with it later.  Email's perfect for this; the other OSM options that I've used (forum/IRC/Telegram/Slack) aren't.

Also - I'm lucky enough to be someone who lives in a country where good internet access is pretty widespread; you don't have to go far in OSM to talk to people who limit their online time or minimise downloaded data column to reduce costs.

**Comment by natrius** [comment]
So, i made a list on what it means to stay with fluxBB or choose something new. 
Discourse: 

- picture-upload
- post-editing with markdown, bbcode and html (not just bbcode)
- reactions on posts (not everybody likes it, but sometimes people want to approve or disapprove a post without posting "+1" all the time)
- mark a post as answer (for help-alike)
- trust-system is available (also for help-alike)
- proper, native mobile support (i made the 'new' theme for the current fluxBB forum and it kinda works, but yeah)
- **proper RTL-language support** for a global community this is crucial! FluxBB did get an pull request that did not got merged for about 4 years now? Don't know, long time. 
- direct answer with e-mail possible
- oauth2 support (tomhughes said already, osm would need to support oauth2 as well)
- integrated updatesystem, not via ftp like fluxbb right now (not useful for a user)
- integrated backup (not useful for a user)
- history for edited posts (that would be sometimes really, really handy)
- and, excluding JS-deniers (how do they use osm btw?)
- migration possible
- constant development

Staying with FluxBB

- **excluding RTL-languages**
- old software without development
- keeping more channels to talk instead of reducing
- no change, so nobody has to 'learn' something new.


The mainreason i am posting now is: french made their own discourse and migrated the existing content already. 
https://forum.openstreetmap.fr/

**Comment by HarelM** [comment]
How can I migrate the Israeli forum like france did? Current communication is mainly telegram...

**Comment by mnalis** [comment]
@natrius While I don't much care about fluxBB vs. Discourse, and I take your word that Discourse is superior here, I do forsee many problems by trying to cram several different communication channels into one. You know the truism: `if it says "one size fits everyone", it doesn't fit anybody`.

Here are broad categories currently in use (by OSM community and wider):

- **Q&A sites** like help.openstreetmap.org: critical thing here is "**best answer on top**". Like StackExchange and friends. Trying to replace it with forum-alike which has "likes", but sorts chronologically instead by likes and not being split into separate questions (no, topics/subjects of some discussion are not the same as questions, generally speaking) is bad idea. Try imagining stackoverflow.com (let's ignore their political problems for the moment) being replaced by Discourse. Wouldn't work at all. (that's why they've taken the throne there - other people were trying to do the Q&A sites with forums+likes+mark_as_answer, which are still really bad by comparison [i.e. _marked as answer_ is often **not** the answer you're looking for]) 

- **chat-like sites** (like Telegram, Slack, Twitter, Facebook, XMPP etc.): they have way too many messages, which are way too little information, and with too little metadata interconnection (eg. missing full references metadata) **to be mixed** with forum/mailing-list like solution. Trying to merge those would quickly drive away one group of users (probably the ones who were using mailing lists).

- **discussion-oriented sites** (like Web Forums - fluxBB&Discourse, Usenet News, Mailing lists, Slashdot, Issue trackers like Github etc). Here I personally prefer:
  -  **mailing lists** for topics I actively follow (completely offline use is the must!) in reasonable mail client which has good support for threading and filtering.
  - **Usenet News client** (via gmane.io) for long threads I only occasionally  need to lookup, due to ability to do reasonable threading and searching subthread for some keywords). 
  - **Github issue tracker** is actually quite usable (as would **Discourse** be, I imagine - I've only actively used one at Let's encrypt) as long as **number of messages in thread is very small** - like dozen or two messages at most. With 50+  messages in thread, all "forums" solutions I've found (Discourse included) become **horrible waste of time** when compared to mailing lists. They lack subthreads that mailing lists/news nicely support so you can't skip over parts which should've been new threads but didn't, they lack reasonable filtering on the subgroup of messages, navigation is just horrible (esp. if it involves who replied to what, limiting/filter by users), they're horribly slow, require online connection etc. All forums suck compared to good e-mail mailing list solution.

Trying to merge more than one group of those is impossible and terrible idea, which is usually self evident quite quickly (sometimes unfortunately only after it is implemented and it's too late to go back - let's try to avoid that future).

But even merging only those discussion-oriented solutions together (which might sometimes be possible - like gmane.io shows for mail/news gatewaying for example) has it's set of problems, some of which some people even see as advantages, like:
  -  `post-editing with markdown, bbcode and html (not just bbcode)` - it's all nice **only** if you can convert them one to another depending on the access method automagically. Eg. please don't force formatting in non-mailing-list-friendly way to mailing list users - eg. I don't read my mails in web browser, so HTML is no-go, and bbcode sucks even more. Simple markdown-alike tagging is OK though (ie. using "*" or "_" around words/phrases to emphasize them is fine if not overdone, but converting plaintext e-mail to XML-alike abomination and claiming "_but we have mailing list support_" is most definitely not)
  - post-editing penalizes e-mail users (as they won't see changes and so confuses the communication), unless you resend messages on each edit, which is bad idea on it's own (as it spams e-mail users if it happens anything but rarely). Also, search should absolutely support finding all historic versions, with appropriate user-defined filters (I've used solutions which only search last revision, and it is extremely frustrating not to be able to find something because someone changed it in the meantime, thus removing critical keyword and hiding needle in the haystack again)
 
>     * and, excluding JS-deniers (how do they use osm btw?)

FYI, I rarely use web browser (much less JavaScript!) to either improve or use OSM, and still use it more efficiently than if I did...

So **TL;DR** - I'm OK with replacing _fluxBB web forum_ with _Discourse web forum_. What I do have a problem with, is the idea "_let's replace all communication channels with Discourse, in order to minimize engineer hours used on supporting the software_". While I'm all for lessening the load on the overloaded engineers (I even try to do PR every blue moon, trying to help there, however small), I do not think that it is good enough reason for doing genocide on major parts of the community in the name of **mistaken idea that "one size fits all".** I'd much rather keep the suboptimal status quo if such outcome is even the possibility.

**Comment by natrius** [comment]
Just to be clear: What needs to be replaced is the forum and QA. Discourse should mainly be the forum. But, it seems like its possible to integrate QA in a nice way. Your critique `[i.e. marked as answer is often not the answer you're looking for]` has nothing to do with the software, it has to do with the people picking the answers. But i guess what you mean with that as well is "put the most upvoted at top" and not "put marked at top". Because marked at top is possible https://meta.discourse.org/t/how-to-mark-a-topic-as-resolved/81793 

Chat is something else and should not get merged with this. Not at all. (in my opinion).

What i'm suggesting is not "Remove everything and lets Discourse do the job": **FluxBB and QA needs to be replaced**. Replacing the forum can be achieved with Discourse. It is possible to integrate QA there as well, but maybe there would be some drawbacks with that, i acknowlege that. Integration with mailinglists would just be the cherry on top because there is massive knowlege around mailinglists and i know quite some new people who just don't want to use a mailinglist. If it does not happen, well, there is basically no change from the current situation: there is a forum and there are mailinglists. Let it be.

**Comment by iandees** [comment]
Tom's work on OAuth in https://github.com/openstreetmap/openstreetmap-website/pull/3177 looks to be merged. Looks like I should be able to test the OAuth 2 connection in my Discourse setup at some point. Has anyone tried that already?

**Comment by mmd-osm** [comment]
Yup, I got it working already.

**Comment by Firefishy** [comment]
Progress update:

1. We have a "private" test instance for Discourse up and running, known as "community". Once testing is done, we will reset the instance and publicly launch.
2. https://github.com/openstreetmap/openstreetmap-website/pull/3300 is to allow seamless sign-in to the discourse insance using osm.org credentials.

**Comment by Firefishy** [comment]
There has been some discussion / planning of the discourse instance in this slack channel: https://osmus.slack.com/archives/C02804ANBC6 ( #discourse-community-rollout ).

**Comment by matkoniecz** [comment]
https://community.openstreetmap.org/ has this new test instance. Is mailing list interface also enabled on this test instance?

(that would be needed for "Merge (...) MLs to discourse?" part)

**Comment by nukeador** [comment]
Hi all,

I want to make sure you are updated about progress here. There are a couple of technical issues (related with https certificates y backups) to solve before going live and the Ops team has been really busy the past months. I'll post it here once we have clarity on the timeline to have this completed, I so sorry for the delay.

Cheers.

**Comment by samwilson** [comment]
I'm not trying to hassle anyone at all, nor increase stress anywhere, but... is there any news about when the new Discourse site might be up and running? :-)

Thanks!

**Comment by cquest** [comment]
> I'm not trying to hassle anyone at all, nor increase stress anywhere, but... is there any news about when the new Discourse site might be up and running? :-)

https://community.openstreetmap.org/ is UP !

**Comment by matkoniecz** [comment]
I admit that I still strongly prefer FluxBB, which has much better interface and is simply working while Discourse seems to be some JS monster.

**Comment by iandees** [comment]
Thanks for everyone working on getting this set up. It feels really fast, I like the options for email communication, and the opportunity for more effective moderation tools.

**Comment by Matthias84** [comment]
As author of this issue, just once more thank you all for your hard work and effort to give your community forum a new modern home :smiley:

**Comment by grischard** [comment]
Linked issues:

#604 for forum migration to discourse
#149 for osqa migration to discourse

**Comment by Firefishy** [comment]
The forum migration was completed under #604.

#149 is for community.openstreetmap.org (discourse) of OSQA.

Mailinglist migrations are happening in different communities by "migrating" their communications over to community.openstreetmap.org (discourse). Migrating the mailinglist archives across to discourse is very unlikely to happen. The old mailinglist post archive will be kept available. Further discussion on this topic is likely best via community.openstreetmap.org

Considering the above, I'm going to close this ticket as completed.

**Comment by Firefishy** [comment]
Started a list of retired mailinglist which can be archived: https://github.com/openstreetmap/operations/issues/975
